pkg/plugin/processor/standalone/host_module.go

```diff
@@ -139,9 +155,137 @@ func (m *hostModuleInstance) commandResponse(ctx context.Context, buf types.Byte
 	if err != nil {
 		m.logger.Err(ctx, err).Msg("failed unmarshalling protobuf command response")
 		m.commandResponses <- tuple[*processorv1.CommandResponse, error]{nil, err}
-		return wasm.ErrorCodeUnknownCommandResponse
+		return pprocutils.ErrorCodeUnknownCommandResponse
 	}
 
 	m.commandResponses <- tuple[*processorv1.CommandResponse, error]{&resp, nil}
 	return 0
 }
+
+// createSchema is the exported function that is called by the WASM module to
+// create a schema and set the buffer bytes to the marshalled response.
+// It returns the response size on success, or an error code on error.
+func (m *hostModuleInstance) createSchema(ctx context.Context, buf types.Bytes) types.Uint32 {
+	m.logger.Trace(ctx).Msg("executing create_schema")
+
+	// check if there's a parked response from the last call
+	if m.parkedCreateSchemaResponse != nil {
+		// make sure the request is the same as the last one
+		if bytes.Equal(buf[:len(m.parkedCreateSchemaBuffer)], m.parkedCreateSchemaBuffer) {
+			out, err := proto.MarshalOptions{}.MarshalAppend(buf[:0], m.parkedCreateSchemaResponse)
+			if err != nil {
+				m.logger.Err(ctx, err).Msg("failed marshalling protobuf create schema response")
+				return pprocutils.ErrorCodeStart
+			}
+			m.parkedCreateSchemaResponse = nil
+			m.parkedCreateSchemaBuffer = nil
+			return types.Uint32(len(out))
+		}
+	}
+
+	var req procutilsv1.CreateSchemaRequest
+	length := len(buf)
+	if m.parkedCreateSchemaBuffer != nil {
+		length = len(m.parkedCreateSchemaBuffer)
+	}
+	err := proto.Unmarshal(buf[:length], &req)
+	if err != nil {
+		m.logger.Err(ctx, err).Msg("failed unmarshalling protobuf create schema request")
+		return pprocutils.ErrorCodeStart
```

Comment:
Is there a reason for not returning a specific error here?

Reply:
because an error about failed unmarshalling or marshaling will not be useful for the user when trying to get or create a schema, this would be a problem from the processor SDK itself or conduit, and will be logged if we got a problem and have to fix it, but the user shouldn't encounter this error.
Other errors that will mean something to the user are the ones returned from "`create`" and "`get`" schema, like an invalid schema or a subject that doesn't exist.

Reply:
I see... So maybe we can then have something like an "internal error"? `ErrorCodeStart` feels a bit unspecific.

Reply:
we'd need to rename it for the processor SDK first, but I agree with the change, will crate a PR for it after this one